src/wasm/asm-wasm-builder.cc

```diff
@@ -434,6 +437,104 @@ class AsmWasmBuilderImpl : public AstVisitor {
     RECURSE(Visit(expr->expression()));
   }
 
+  bool TestIntBinaryOperation(BinaryOperation* expr, 
```

Comment:
These TestXX methods should be called MatchXXX, since they do pattern matching on the AST.